### PR TITLE
[saasherder] GQL classes and type hints

### PR DESCRIPTION
**Pull request by chassing**
Refactor `saasherder` to support generated GQL classes and add type hints

Ticket: [APPSRE-7152](https://issues.redhat.com/browse/APPSRE-7152)

### CODE REVIEW

**Comment by chassing** [comment]
@maorfr, you know `saasherder` at most. Can I kindly ask you to review this PR 🙇

**Review by maorfr — APPROVED** [review]

File: reconcile/gql_definitions/common/saas_files.gql [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+      }
+    }
+    deployResources {
+      requests {
+        cpu
+        memory
+      }
+      limits {
+        cpu
+        memory
```
Comment (fishi0x01) [inline reconcile/gql_definitions/common/saas_files.gql:42]: nit: I think we have a fragment for these https://github.com/app-sre/qontract-reconcile/blob/master/reconcile/gql_definitions/fragments/resource_values.gql

**Review by fishi0x01 — APPROVED** [review]
Awesome!!! :heart:

File: reconcile/gql_definitions/common/saas_files.gql (hunk 1 above)
Comment (chassing) [reply reconcile/gql_definitions/common/saas_files.gql:42]: Unfortunately, the fragment is of type `ResourceValues_v1`, and this type is `ResourceRequirements_v1`. Both are very similar, but the attributes are required in `ResourceRequirements_v1`.

**Pull request merged**